conjur/user/user_logic.py

```diff
@@ -0,0 +1,100 @@
+
+# Internals
+from conjur.credentials_data import CredentialsData
+
+class UserLogic:
+    """
+    UserLogic
+
+    This class holds the business logic for handling
+    user activity
+    """
+    def __init__(self, conjurrc_data, credentials_from_store, client, resource):
+        self.conjurrc_data=conjurrc_data
+        self.credentials_from_store=credentials_from_store
+        self.client=client
+        self.resource=resource
+
+    # pylint: disable=logging-fstring-interpolation
+    def rotate_api_key(self, resource_to_update):
+        """
+        Method to trigger two types of API key actions.
+        1. The user can either rotate their own API key or
+        2. Rotate another users API key
+        """
+        loaded_credentials = self.extract_credentials_from_credential_store()
+        user_from_credential_store = loaded_credentials['login_id']
+        if resource_to_update == user_from_credential_store:
+            raise Exception("Error: To rotate the API key of the currently logged-in user "\
+                            " use this command without any flags or options")
+
+        if resource_to_update is not None:
+            user_from_credential_store = resource_to_update
+            logging.debug(f"Rotating API key for '{user_from_credential_store}'")
+            new_api_key = self.rotate_anothers_api_key(user_from_credential_store)
+            logging.debug(f"API key for '{user_from_credential_store}' successfully rotated")
+        else:
+            # if the user does not provide a user to rotate their API key,
+            # their own API key will be rotated
+            logging.debug(f"Rotating API key for '{user_from_credential_store}'")
+            new_api_key = self.rotate_personal_api_key(user_from_credential_store,
+                                                       loaded_credentials['api_key'])
+
+            # Update the new rotated API for the logged-in user
+            logging.debug("Updating credential store with new API key " \
+                          f"for '{user_from_credential_store}'")
+            self.update_api_key_in_credential_store(user_from_credential_store,
+                                                    loaded_credentials,
+                                                    new_api_key)
+
+        return user_from_credential_store, new_api_key
+
+    # pylint: disable=logging-fstring-interpolation
+    def change_password(self, new_password):
+        """
+        Method to call the client to change the logged in user's password
+        """
+        loaded_credentials = self.extract_credentials_from_credential_store()
+        resource_to_update = loaded_credentials['login_id']
+        logging.debug(f"Changing password for '{resource_to_update}'")
+        # pylint: disable=line-too-long
+        return self.client.change_password(resource_to_update, loaded_credentials['api_key'], new_password)
+
+    def extract_credentials_from_credential_store(self):
```

Comment:
goo encapsulation. all conjurrc and netrc action are better to be encapsulate as we might change the way we save credential in the future